assets/app/scripts/services/apiService.js

```diff
@@ -0,0 +1,268 @@
+        };
+      }
+
+      // else, legacy
+      var found = _.find(API_CFG, function(api) {
+        return api.resources[_.first(resource.split('/'))] || api.resources['*']; // ['*'] ?
+      });
+
+      return found ?
+              {
+                hostPort: found.hostPort,
+                prefix: found.prefixes[qualified.version] ||
+                        found.prefixes[defaultVersionForGroup[qualified.group]] ||
+                        found.prefixes['*'],
+                version: qualified.version || defaultVersionForGroup[qualified.group]
+              } :
+              undefined;
+    };
+
+
+    // TODO: improve this when we have API discovery...
+    // takes: {resource: '', group:'', verison: ''} or string "pods"
+    // if string, will result in a v1 api, oapi response
+    var apiExistsFor = function(unqualified) {
+      var qualified = qualifyResource(unqualified);
+
+      // if we have a group, this is the best we can do until we get discovery.
+      // we have no way to know if the resources really exists at the endpoint at this time.
+      if(qualified.group === '') {
+        return !!findAPIFor(qualified);
+      }
+      return true;
+    };
+
+
+    // URITemplate has no ability to do conditional {+group}, so templates are now broken up
+    var API_TEMPLATE = "{protocol}://{+hostPort}{+prefix}/{version}/";
+    var API_GROUP_TEMPLATE = "{protocol}://{+hostPort}{+prefix}/{+group}/{version}/";
+
+    var URL_GET_LIST              = "{resource}{?q*}";
+    var URL_OBJECT                = "{resource}/{name}{/subresource*}{?q*}";
+    var URL_NAMESPACED_GET_LIST   = "namespaces/{namespace}/{resource}{?q*}";
+    var URL_NAMESPACED_OBJECT     = "namespaces/{namespace}/{resource}/{name}{/subresource*}{?q*}";
+
+
+    var findTemplateFor = function(name, namespace, group) {
+      var base = group ? API_GROUP_TEMPLATE : API_TEMPLATE;
+      var rest = namespace ? URL_NAMESPACED_GET_LIST : URL_GET_LIST;
+      if(name) {
+        rest = namespace ? URL_NAMESPACED_OBJECT : URL_OBJECT;
+      }
+      return (base + rest);
+    };
+
+
+    var protocol = function(isWebsocket) {
+      if(isWebsocket) {
+        return $window.location.protocol === "http:" ?  "ws" : "wss";
+      }
+      return $window.location.protocol === "http:" ? "http" : "https";
+    };
+
+
+    var cleanCopyParams = function(params) {
+      params = (params &&  angular.copy(params)) || {};
+      delete params.namespace;
+      return params;
+    };
+
+
+    // generates something like:
+    // https://localhost:8443
+    var openshiftAPIBaseUrl = function() {
+      var protocol = $window.location.protocol === "http:" ? "http" : "https";
+      var hostPort = API_CFG.openshift.hostPort;
+      return new URI({protocol: protocol, hostname: hostPort}).toString();
+    };
+
+    // TODO: can this be implmeneted in a meangful way at this point?
+    var resourceInfo = function(resource) {
```

Comment:
I think all calls to this are either `apiExistsFor` or `findAPIFor`... qualify at the call site and remove this func